src/lib/utils/contractUtils.js

```diff
@@ -0,0 +1,44 @@
+import { StrKey, Contract } from '@stellar/stellar-sdk'
+import { Server } from '@stellar/stellar-sdk/rpc'
+
+/**
+ * Generates a contract client for interacting with a Stellar smart contract
+ * @param {string} contractId - The contract ID to validate and connect to
+ * @returns {Promise<Contract>} A Contract instance
+ */
+export async function generateContractClient(contractId) {
+    const server = new Server('https://soroban-testnet.stellar.org')
+
+    // Validate contract ID
+    if (!StrKey.isValidContract(contractId)) {
+        throw new Error('Invalid contract ID format')
+    }
+
+    try {
+        console.log('Fetching WASM for contract ID:', contractId)
+
+        // Fetch the contract's WASM using the contract ID
+        const contractResponse = await server.getContractWasmByContractId(contractId)
+
+        // Log the WASM bytecode length or the data for debugging
+        console.log('Contract WASM length:', contractResponse.length) // Adjust if needed based on actual response structure
+
+        // Create a contract instance (you may need to initialize this differently based on your library's requirements)
+        const contract = new Contract(contractId)
+
+        // You might want to add the contract's interface here
+        // This could involve parsing the WASM to get available methods
+
+        return contract // Return the contract instance
+    } catch (serverError) {
```

Comment:
When I first tried to load a contract locally, I used a SAC contract (they don't have a Wasm executable associated with them, since they're just "built-in"). I imagine this would be a pretty common situation for users of this functionality. Let's add a check somewhere to handle this case and build a SAC client for that asset contract.